app/crud/category.py

```diff
@@ -0,0 +1,60 @@
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy import select, delete
+from uuid import UUID
+from typing import List, Optional
+
+from app.models.category import Category
+from app.schemas.category import CategoryCreate, CategoryUpdate
+
+
+async def create_category(db: AsyncSession, category_data: CategoryCreate) -> Category:
+    db_category = Category(
+        category_name=category_data.category_name,
+        category_color=category_data.category_color,
+    )
+    db.add(db_category)
+    await db.commit()
+    await db.refresh(db_category)
+    return db_category
+
+
+async def read_categories(db: AsyncSession) -> List[Category]:
+    result = await db.execute(select(Category))
+    return result.scalars().all()
+
+
+async def read_category_by_name(
+    db: AsyncSession, category_name: str
+) -> Optional[Category]:
+    result = await db.execute(
+        select(Category).where(Category.category_name == category_name)
+    )
+    return result.scalars().first()
+
+
+async def read_category_by_id(
+    db: AsyncSession, category_id: UUID
+) -> Optional[Category]:
+    result = await db.execute(select(Category).where(Category.id == category_id))
+    return result.scalars().first()
+
+
+async def update_category(
+    db: AsyncSession, category_id: UUID, category_data: CategoryUpdate
+) -> Category:
+    db_category = await read_category_by_id(db, category_id)
+
+    update_data = category_data.model_dump(exclude_unset=True)
+
+    for key, value in update_data.items():
+        setattr(db_category, key, value)
+
+    await db.commit()
+    await db.refresh(db_category)
```

Comment:
DB update에 문제가 발생할 수도 있을 것 같은데, 문제 생기면 rollback해야하지 않을까용?

@trivialcoding6  의견주세용

```suggestion
    try:
        await db.commit()
        await db.refresh(db_category)
    except Exception as e:
        await db.rollback()
        raise e
```

Reply:
@GideokKim 
저같은 경우는 단일 요청에서 디비 업데이트에 문제가 생겼다는건 업데이트가 안된거라는 의미로 봐서 따로 롤백은 넣지 않긴했거든용 
예를들어 특정 테이블 먼저 수정 후 다른 테이블도 수정이 이루어져야 할때 먼저는 성공하고 이후네 실패할 경우에는 롤백이 필요해 보이는데 단일 요청에서는 필요성이 있을지는 정확히 모르겠네요..! 필요한 상황이 있다면 있으면 안전할거 같긴합니당 :)

Reply:
@trivialcoding6 들어보니 필요없을 것 같긴하네용 resolve 처리하겠습니다 ㅎㅎ